File: src/main/java/guru/xgm/language/lni/formatting/panel/LniAlignTokenPanel.java
package guru.xgm.language.lni.formatting.panel;

import com.intellij.application.options.codeStyle.OptionTreeWithPreviewPanel;
import com.intellij.lang.Language;
import com.intellij.openapi.fileTypes.FileType;
import com.intellij.openapi.util.NlsContexts;
import com.intellij.psi.codeStyle.CodeStyleSettings;
import com.intellij.psi.codeStyle.LanguageCodeStyleSettingsProvider;
import guru.xgm.language.lni.lang.LniLanguage;
import guru.xgm.language.lni.openapi.fileTypes.LniFileType;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import javax.swing.*;

public class LniAlignTokenPanel extends OptionTreeWithPreviewPanel {
    public LniAlignTokenPanel(CodeStyleSettings settings) {
        super(settings);
        init();
    }

    @NlsContexts.Label
    public static String GROUP_PROPERTY = "Property";

    @NlsContexts.Label
    public static String GROUP_NATIVE_DECL = "Native declaration";

    @NlsContexts.Label
    public static String GROUP_GVAR = "Global variable";

    @Override
    protected void initTables() {
        initCustomOptions(GROUP_PROPERTY);
        initCustomOptions(GROUP_NATIVE_DECL);
        initCustomOptions(GROUP_GVAR);
    }

    @Override
    public JComponent getPanel() {
        return myPanel;
    }

    @Override
    protected @NlsContexts.TabTitle @NotNull String getTabTitle() {
        return "Align Token";
    }

    @Override
    public LanguageCodeStyleSettingsProvider.SettingsType getSettingsType() {
        return LanguageCodeStyleSettingsProvider.SettingsType.LANGUAGE_SPECIFIC;
    }

    @Override
    protected void customizeSettings() {
        LanguageCodeStyleSettingsProvider provider = LanguageCodeStyleSettingsProvider.forLanguage(LniLanguage.INSTANCE);
        if (provider != null) provider.customizeSettings(this, getSettingsType());
    }

    @Override
    @NotNull
    protected final FileType getFileType() {
        return LniFileType.Companion.getInstance();
    }

    @Nullable
    @Override
    public Language getDefaultLanguage() {
        return LniLanguage.INSTANCE;
    }
}

File: src/main/kotlin/raft/war/language/lni/formatting/panel/LniAlignTokenPanel.kt
package raft.war.language.lni.formatting.panel

import com.intellij.application.options.codeStyle.OptionTreeWithPreviewPanel
import com.intellij.lang.Language
import com.intellij.openapi.fileTypes.FileType
import com.intellij.openapi.util.NlsContexts
import com.intellij.openapi.util.NlsContexts.TabTitle
import com.intellij.psi.codeStyle.CodeStyleSettings
import com.intellij.psi.codeStyle.LanguageCodeStyleSettingsProvider
import com.intellij.psi.codeStyle.LanguageCodeStyleSettingsProvider.SettingsType
import raft.war.language.lni.lang.LniLanguage
import raft.war.language.lni.openapi.fileTypes.LniFileType

class LniAlignTokenPanel(settings: CodeStyleSettings?) : OptionTreeWithPreviewPanel(settings) {
    init {
        init()
    }

    override fun initTables() {
        initCustomOptions(GROUP_PROPERTY)
        initCustomOptions(GROUP_NATIVE_DECL)
        initCustomOptions(GROUP_GVAR)
    }

    override fun getTabTitle(): @TabTitle String = "Align Token"
    override fun getSettingsType(): SettingsType = SettingsType.LANGUAGE_SPECIFIC
    override fun getFileType(): FileType = LniFileType.instance
    override fun getDefaultLanguage(): Language = LniLanguage.instance

    override fun customizeSettings() {
        val provider = LanguageCodeStyleSettingsProvider.forLanguage(LniLanguage.instance)
        provider?.customizeSettings(this, settingsType)
    }

    companion object {
        var GROUP_PROPERTY: @NlsContexts.Label String? = "Property"
        var GROUP_NATIVE_DECL: @NlsContexts.Label String? = "Native declaration"
        var GROUP_GVAR: @NlsContexts.Label String? = "Global variable"
    }
}
